### PR TITLE
Experimental versions

### LINKED ISSUE #26: Support experimental versions
It'd be cool to be able to install and flip between experimental versions, too. Installation instructions are [in the Docker repo](https://github.com/docker/docker/tree/master/experimental).

`dvm ls-remote` wouldn't pick these up, but maybe that's okay.

### PR DESCRIPTION
**Pull request by rgbkrk**
Support experimental Docker binaries. Closes #26.

### CODE REVIEW

**Comment by rgbkrk** [comment]
I'll let @thomasem or @smashwilson give this a pass before merging.

**Comment by rgbkrk** [comment]
```
$ dvm install experimental
experimental is already installed, but we'll assume you want the latest experimental
######################################################################## 100.0%
Now using Docker experimental
$ dvm use experimental
Now using Docker experimental
$ docker --version
Docker version 1.9.0-dev, build 6654b0e, experimental
```

File: dvm.sh [hunk 1]
```diff
@@ -636,10 +650,36 @@ dvm() {
...
+            DOCKER_BINARY="docker"
+            DVM_ARCH="$(dvm_get_arch)"
+            DVM_OS="$(dvm_get_os)"
+
+            checksum_url="$DVM_GET_EXPERIMENTAL_DOCKER_MIRROR/$DVM_OS/$DVM_ARCH/docker-latest.sha256"
+
+            local sum
+            sum=`dvm_download -L -s $checksum_url -o - | command awk '{print $1}'`
+
+            if ( dvm_checksum "$VERSION_PATH/docker" $sum ); then
```
Comment (smashwilson) [inline dvm.sh]: I'm assuming `dvm_checksum` is already pretty portable? I know getting `sha256sum` to work the same across OS X and Linux is a bit of a pain.

**Comment by smashwilson** [comment]
Looks pretty good to me!

File: dvm.sh (hunk 1 above)
Comment (rgbkrk) [inline dvm.sh]: As far as I know. It's roughly adapted from nvm's but with a different hashing algorithm.

**Pull request merged**